### PR TITLE
feat: replace unmaintained winapi with windows-sys

### PR DESCRIPTION
**Pull request by panekj**
winapi has been unmaintained for a long time
this commit replaces winapi with windows-sys which is generated from windows API headers and maintained by Microsoft
apart from that, codebase has been fixed from few clippy lints and errors

winapi is still an indirect dependency via:
- [ ] [atty](https://github.com/softprops/atty) | https://github.com/softprops/atty/pull/54
- [ ] [term](https://github.com/Stebalien/term)
- [ ] [dir-sys-next](https://github.com/xdg-rs/dirs) | https://github.com/xdg-rs/dirs/pull/62

### CODE REVIEW

**Comment by panekj** [comment]
@joaomoreno what's the proper way to test all functionality?
I'll admit I'm not an expert in writing safe code
I've been able to check only `--gui` parameter
image

**Comment by panekj** [comment]
I've also tried to maintain preexisting style of codebase and added rustfmt config that preserves it

**Comment by panekj** [comment]
I'm also not sure what Rust version you use (CI seems to use stable) so I've updated code to latest rust compatibility (1.64, it will fail for anything older)

**Comment by joaomoreno** [comment]
> I've also tried to maintain preexisting style of codebase and added rustfmt config that preserves it

Feel free to submit a different PR which brings in rustfmt with more modern rules.

**Comment by joaomoreno** [comment]
> @joaomoreno what's the proper way to test all functionality? I'll admit I'm not an expert in writing safe code I've been able to check only `--gui` parameter image

The win32 specific part is purely related to GUI, so manual testing with `--gui` is good enough.

File: src/main.rs [hunk 1]
```diff
@@ -100,17 +100,16 @@ fn delete_existing_version(
 	let root = PathBuf::from(root_path);
 	directories.push_back(root);
 
-	while directories.len() > 0 {
+	while directories.is_empty() {
```
Comment (joaomoreno) [inline src/main.rs:103]: Whoops.

File: src/main.rs [hunk 2]
```diff
@@ -396,32 +400,28 @@ fn _main(log: &slog::Logger, args: &Vec<String>) -> Result<(), Box<dyn error::Er
...
-	msgs.push("Please read the log file for more information:");
-	msgs.push(log_path);
-
-	let msg = msgs.join("\n\n");
+	let msg = format!(
+		"Failed to install Visual Studio Code update.
+		Updates may fail due to anti-virus software and/or runaway processes. Please try restarting your machine before attempting to update again.
+		Please read the log file for more information:
+		{log_path}"
+	);
```
Comment (joaomoreno) [inline src/main.rs:419]: This will add weird indentation to the message.

File: src/process.rs [hunk 3]
```diff
@@ -1,235 +1,229 @@
...
+			cntThreads: 0,
+			th32ParentProcessID: 0,
+			pcPriClassBase: 0,
+			dwFlags: 0,
+			szExeFile: [0u16; 260],
+		};
+
+		pe32.dwSize = mem::size_of::<PROCESSENTRY32W>() as u32;
+
+		if Process32FirstW(handle, &mut pe32).is_negative() {
```
Comment (joaomoreno) [inline src/process.rs:49]: This breaks the logic: BOOL is 1 for true and 0 for false, while `is_negative()` catches neither case.

File: src/process.rs [hunk 4]
```diff
@@ -1,235 +1,229 @@
...
+		loop {
+			result.push(RunningProcess {
+				name: from_utf16(&pe32.szExeFile).map_err(|e| {
+					CloseHandle(handle);
+					e
+				})?,
+				id: pe32.th32ProcessID,
+			});
+
+			if Process32NextW(handle, &mut pe32).is_negative() {
```
Comment (joaomoreno) [inline src/process.rs:69]: This breaks the logic: BOOL is 1 for true and 0 for false, while `is_negative()` catches neither case.

**Review by joaomoreno — APPROVED** [review]

**Review by sandy081 — APPROVED** [review]

**Pull request merged**